Add PRW blog post (#1943)
prometheus/docs: content/blog/2021-05-04-prometheus-conformance-remote-write-compliance.md

---

> title: "Prometheus conformance program: Remote Write Compliance test results"

Reviewer: We do title casing for blog post titles, so this would be:
```suggestion
title: "Prometheus Conformance Program: Remote Write Compliance Test Results"
```

created_at: 2021-05-05
kind: article
author_name: Richard "RichiH" Hartmann
---

As [announced by CNCF](https://www.cncf.io/blog/2021/05/03/announcing-the-intent-to-form-the-prometheus-conformance-program/) and by [ourselves](https://prometheus.io/blog/2021/05/03/introducing-prometheus-conformance-program/), we're starting a Prometheus conformance program.

> To give everyone an overview of where the ecosystem is before running tests officially, we wanted to show off the newest addition to our happy little bunch of test suites: The Prometheus Remote Write tests.

Reviewer: Add a short sentence at the end of this paragraph explaining what Remote Write is?

Reviewer: Also, that this is testing only the sending end, not the receiving end?

Author: doine

> During Monday's [PromCon](https://promcon.io/2021-online/), Tom presented the test results from the time of recording a few weeks ago. In the live section, he already had an [update](https://docs.google.com/presentation/d/1RcN58LlS3V5tYCUsftqUvNuCpCsgGR2P7-GoH1MVL0Q/edit#slide=id.gd1789c7f7c_0_0). Two days later we have two more updates: The addition of [vector](https://github.com/prometheus/compliance/pull/24) and [new versions](https://github.com/prometheus/compliance/pull/25).

Reviewer: "Tom" -> "Tom Wilkie", and link to him

Author: done

> During Monday's [PromCon](https://promcon.io/2021-online/), Tom presented the test results from the time of recording a few weeks ago. In the live section, he already had an [update](https://docs.google.com/presentation/d/1RcN58LlS3V5tYCUsftqUvNuCpCsgGR2P7-GoH1MVL0Q/edit#slide=id.gd1789c7f7c_0_0). Two days later we have two more updates: The addition of [vector](https://github.com/prometheus/compliance/pull/24) and [new versions](https://github.com/prometheus/compliance/pull/25).

Reviewer: "vector" is called "Vector" with a capital V.

I'd not require people to click on the link to find out what Vector is. How about:

"The addition of the observability pipeline tool Vector, as well as new versions of existing systems."

Author: done

So, without further ado, the current results in alphabetical order are:

> * Grafana Agent 0.13.1: 100%

Reviewer: I'd make a Markdown table out of this, since it's not quickly readable otherwise. See https://github.com/prometheus/docs/blob/master/content/docs/practices/histograms.md#quantiles for an example of a table.

* Prometheus 2.26.0: 100%
* OpenTelemetry Collector 0.26.0: 41%
* Telegraf 1.18.2: 65%
* Timber Vector 0.13.1: 35%
* VictoriaMetrics Agent 1.59.0: 76%

> The raw results are

Reviewer: ```suggestion
The raw results are:
```

    --- PASS: TestRemoteWrite/vector/SortedLabels (10.02s)
    --- PASS: TestRemoteWrite/vector/Summary (10.02s)
    --- PASS: TestRemoteWrite/vector/Timestamp (10.02s)
--- FAIL: TestRemoteWrite/vmagent (0.01s)
    --- FAIL: TestRemoteWrite/vmagent/Invalid (20.66s)
    --- FAIL: TestRemoteWrite/vmagent/Ordering (22.05s)
    --- FAIL: TestRemoteWrite/vmagent/RepeatedLabels (20.67s)
    --- FAIL: TestRemoteWrite/vmagent/Staleness (20.67s)
    --- PASS: TestRemoteWrite/vmagent/Counter (20.67s)
    --- PASS: TestRemoteWrite/vmagent/EmptyLabels (20.64s)
    --- PASS: TestRemoteWrite/vmagent/Gauge (20.66s)
    --- PASS: TestRemoteWrite/vmagent/Headers (20.64s)
    --- PASS: TestRemoteWrite/vmagent/Histogram (20.66s)
    --- PASS: TestRemoteWrite/vmagent/HonorLabels (20.66s)
    --- PASS: TestRemoteWrite/vmagent/InstanceLabel (20.66s)
    --- PASS: TestRemoteWrite/vmagent/JobLabel (20.66s)
    --- PASS: TestRemoteWrite/vmagent/NameLabel (20.66s)
    --- PASS: TestRemoteWrite/vmagent/SortedLabels (20.66s)
    --- PASS: TestRemoteWrite/vmagent/Summary (20.66s)
    --- PASS: TestRemoteWrite/vmagent/Timestamp (20.67s)
    --- PASS: TestRemoteWrite/vmagent/Up (20.66s)

> ````

Reviewer: Add at least a one-sentence closer to this after the code block? Like how this is awesome, and stay tuned for further updates, or something.

Author: doen

---
title: "Prometheus Conformance Program: Remote Write Compliance Test Results"
created_at: 2021-05-05
kind: article
author_name: Richard "RichiH" Hartmann
---

As [announced by CNCF](https://www.cncf.io/blog/2021/05/03/announcing-the-intent-to-form-the-prometheus-conformance-program/) and by [ourselves](https://prometheus.io/blog/2021/05/03/introducing-prometheus-conformance-program/), we're starting a Prometheus conformance program.

> To give everyone an overview of where the ecosystem is before running tests officially, we wanted to show off the newest addition to our happy little bunch of test suites: The Prometheus [Remote Write](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write) tests sending of the protocol against our [specification](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM).

Reviewer: ```suggestion
To give everyone an overview of where the ecosystem is before running tests officially, we wanted to show off the newest addition to our happy little bunch of test suites: The Prometheus [Remote Write](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write) compliance test suite tests the sender part of the Remote Write protocol against our [specification](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM).
```
